cdap-ui/app/cdap/components/AbstractWidget/DatasetSelectorWidget/index.tsx

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+import withStyles, { WithStyles, StyleRules } from '@material-ui/core/styles/withStyles';
+import { MyDatasetApi } from 'api/dataset';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import TextField from '@material-ui/core/TextField';
+import If from 'components/If';
+import classnames from 'classnames';
+import ee from 'event-emitter';
+import { objectQuery } from 'services/helpers';
+import ThemeWrapper from 'components/ThemeWrapper';
+import PropTypes from 'prop-types';
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+
+const styles = (theme): StyleRules => {
+  return {
+    textInput: {
+      padding: '10px',
+    },
+    root: {
+      '& $notchedOutline': {
+        borderColor: theme.palette.grey['300'],
+      },
+      '&:hover $notchedOutline': {
+        borderColor: theme.palette.grey['300'],
+      },
+      '&$focused $notchedOutline': {
+        border: `1px solid ${theme.palette.blue[100]}`,
+      },
+    },
+    focused: {},
+    notchedOutline: {}, // empty object to override styles
+    autocompleteContainer: {
+      position: 'relative',
+    },
+    autocomplete: {
+      position: 'absolute',
+      top: 0,
+      left: 0,
+      right: 0,
+      backgroundColor: 'white',
+      border: `1px solid ${theme.palette.grey[400]}`,
+      zIndex: 101,
+      padding: '5px 0',
+      boxShadow: `0 3px 10px 0 ${theme.palette.grey[200]}`,
+    },
+    autocompleteRow: {
+      lineHeight: '1.5rem',
+      padding: '8px 10px',
+      cursor: 'pointer',
+      '&:not(:last-child)': {
+        borderBottom: `1px solid ${theme.palette.grey[400]}`,
+      },
+      '&:hover': {
+        backgroundColor: theme.palette.grey[600],
+      },
+    },
+    highlight: {
+      backgroundColor: `${theme.palette.blue[100]} !important`,
+      color: 'white',
+    },
+  };
+};
+
+interface IDatasetSelector extends WithStyles<typeof styles> {
+  onChange: (value) => void;
+  value: string;
+  placeholder?: string;
+}
+
+enum KEY_CODE {
```

Comment:
nit: Might be worth extracting this out to to a common module. We use this in a few places.. At least the ENTER and ESC in dataprep and modals.

Reply:
i've put it on global-constants file. Will refactor other codes separately